lib/pbench/server/auth/__init__.py

```diff
@@ -0,0 +1,435 @@
+import logging
+from http import HTTPStatus
+from typing import Dict, List, Optional, Union
+from urllib.parse import urljoin
+
+import jwt
+import requests
+from flask_restful import abort
+from requests.structures import CaseInsensitiveDict
+
+from pbench.server import JSON
+from pbench.server.auth.exceptions import OidcError
+
+
+class OpenIDClient:
+    """
+    OpenID client object.
+    :param server_url: Keycloak server url
+    :param client_id: client id
+    :param realm_name: realm name
+    :param client_secret_key: client secret key
+    :param verify: True if want check connection SSL
+    :param headers: dict of custom header to pass to each HTML request
+    """
+
+    AUTHORIZATION_ENDPOINT: str = ""
+    TOKEN_ENDPOINT: str = ""
+    USERINFO_ENDPOINT: str = ""
+    REVOCATION_ENDPOINT: str = ""
+    JWKS_URI: str = ""
+    LOGOUT_ENDPOINT: str = ""
+
+    def __init__(
+        self,
+        server_url: str,
+        client_id: str,
+        logger: logging.Logger,
+        realm_name: str = None,
+        client_secret_key: str = None,
+        verify: bool = True,
+        headers: Optional[Dict[str, str]] = None,
+    ):
+        self.server_url = server_url
+        self.client_id = client_id
+        self.client_secret_key = client_secret_key
+        self.realm_name = realm_name
+        self.logger = logger
+        self.headers = (
+            CaseInsensitiveDict()
+            if headers is None
+            else CaseInsensitiveDict().update(headers)
+        )
+        self.verify = verify
+        self.connection = requests.session()
+        self.set_well_known_endpoints()
+
+    def get_header_param(self, key: str) -> str:
+        """
+        Return a specific header parameter value.
+        Args:
+            key: header key
+        Returns:
+             Header parameters value.
+        """
+        return self.headers.get(key)
+
+    def add_header_param(self, key: str, value: str):
+        """
+        Add a single parameter inside the header.
+        Args:
+            key: Header parameters key.
+            value: Value to be added for the key.
+        """
+        self.headers[key] = value
+
+    def del_param_headers(self, key: str):
+        """Remove a specific header parameter.
```

Comment:
This line has the text on the same line as the triple-quote; however, unlike this and the subsequent function, the preceding functions have the triple-quote on its own line.  You should pick one approach and be consistent.   (PEP-8 and PEP-257 seem to be ambivalent on which is best, although they recommend leaving a blank line after the first, "summary" line.)